src/crypto/algorithms/megolm.js

```diff
@@ -192,7 +210,7 @@ utils.inherits(MegolmEncryption, EncryptionAlgorithm);
  *    OutboundSessionInfo when setup is complete.
  */
 MegolmEncryption.prototype._ensureOutboundSession = async function(
-    devicesInRoom, blocked, singleOlmCreationPhase,
+    room, devicesInRoom, blocked, singleOlmCreationPhase,
```

Comment:
Would it make sense for `EncryptionAlgorithm` to hold the room model from the start at construction time, rather than expecting in certain functions...? It already has the room ID. Just an idea, perhaps there's a reason to prefer this instead.

Reply:
Yeah, I don't know if there's a particular reason why the `EncryptionAlgorithm` only holds the room ID rather than the whole room model.  But I wanted to avoid making too many unrelated changes.